[processing] 2 more plots added to the collection

ghtmtt opened the pull request:
This PR re-uses the code of VectorLayerScatterPlot to make a 3D plot  and adds a simple BoxPlot

ghtmtt committed:
[processing] 2 more plots added to the collection

ghtmtt committed:
fix indentation

ghtmtt committed:
fix indentation 2

ghtmtt committed:
fix indentation 3

m-kuhn commented:
Do you think you could add unit tests for these?

Something like https://github.com/qgis/QGIS/blob/master/python/plugins/processing/tests/testdata/qgis_algorithm_tests.yaml#L1814-L1824

ghtmtt commented:
mmm.. really don't not how to deal with the html output file and run an unit test..  the html file is super huge and with a lot of default values and strings.. (e.g. `!function(t){if("object"==typeof exports&&"undefined"!=typeof module)module.exports=t()` and it continues for thousands of lines).. do you have an idea?

ghtmtt committed:
allow to read string as X variable

ghtmtt commented:
@alexbruy I noticed something wrong with the x variable in the BarPlot (related to numpy, the resulting x axis does not show the correct range).
I also noticed that strings were not allowed in the x variable (and discovered that the method of Processing https://github.com/qgis/QGIS/blob/master/python/plugins/processing/tools/vector.py#L190 automatically converts fields into floats).
Could you have a look at the super small fixes made?

Thanks!

m-kuhn commented:
You could look for some strings which are visible in the end result (like axis labels etc) and regex for them. Just some general safety check if there is output generated and if it contains some of the desired result. This will at least control that a) the algorithm runs fine (no exceptions thrown etc), b) there is an output produced and c) that the output contains "something probably useful".

ghtmtt commented:
Totally agree with a kind of general check that assures the output. I will try to do something.. Let you now

ghtmtt commented:
@m-kuhn  I was not able to write a test, sorry.. can we merge it anyway? I'll look to write some test..
or better, @alexbruy can we use your new https://github.com/qgis/QGIS/pull/4268 test suite for the plots?

alexbruy committed:
[FEATURE] new custom widget — combobox with checkable items

alexbruy committed:
add Python bindings

alexbruy committed:
custom widget for Qt Designer

alexbruy committed:
add tests

nyalldawson committed:
Refactor QgsPalettedRasterRenderer to allow "gaps" in pixel values

Previously the renderer required that pixel values followed
sequential numerical order. This refactor allows values to
be skipped, so that only certain color index will be rendered.

nyalldawson committed:
Rework paletted raster renderer widget

- Switch to a real model
- Allow adding new rows and deleting multiple selected rows
- Allow changing colors inline in style dock (no modal color picker)

nyalldawson committed:
Allow changing pixel value in paletted renderer widget

nyalldawson committed:
Allow changing label for multiple values at once in paletted renderer widget

nyalldawson committed:
[FEATURE] Add a color ramp combo to paletted raster renderer

Allows rapid recoloring of palette to match a color ramp

nyalldawson committed:
Enable reordering of paletted raster entries

alexbruy committed:
more doxymentation

alexbruy committed:
remove private member initialization

alexbruy committed:
make model and delegate classes private API

nyalldawson committed:
Add method to read paletted color data from string

nyalldawson committed:
Add method to load paletted renderer classes from file

nyalldawson committed:
[FEATURE] Allow loading paletted color table from file

Supports standard GDAL/ESRI clr file format

nyalldawson committed:
[FEATURE] Implement saving paletted raster renderer color tables

Use the .clr/gdal file format, but add the labels on the ends of
the lines. Seems other importers like ArcMap just ignore these.

nyalldawson committed:
No lies!

nyalldawson committed:
[FEATURE] Allow classifying paletted renderer using unique values
from a raster layer

Adds an easy way to style discrete rasters such as landuse classes
using the Paletted renderer. Just select the Paletted renderer,
pick a band, then hit the "Add Unique Values" button. The unique
pixel values will be fetched from the layer and a color assigned
to each using the currently selected color ramp.

Fix #14845

Sponsored by
- Stéphane Henriod
- Satelligence (http://satelligence.com/)
- Bird's Eye View (https://www.birdseyeviewgis.com/)
- + other anonymous backers

nyalldawson committed:
[FEATURE] Add button to paletted renderer widget to allow loading
embedded color table

Only enabled for rasters which contain a color table

nyalldawson committed:
Allow cancelation of raster stats/histogram operations

nyalldawson committed:
Calculate unique raster values in background thread

To keep UI nice and responsive

nyalldawson committed:
Fix some crashes in paletted raster renderer

- handle negative color values
- don't crash when calculating unique values in bad rasters

nyalldawson committed:
Add unit test to ensure that #13263 is fixed

Fix #13263

nyalldawson committed:
Auto load raster color table from band when band is changed in
paletted raster renderer

(Only if raster has an embedded color table)

nyalldawson committed:
[needs-docs] Rename Paletted raster renderer to "Paletted/Unique values"

nyalldawson committed:
Default to random color ramps for unique value rasters

nyalldawson committed:
Improve random color generation for unique raster values

nyalldawson committed:
Don't set both a default ramp and show the random option

nyalldawson committed:
Fix classification test

nyalldawson committed:
Add label for color ramp widget

nyalldawson committed:
Simple c++ base class for processing algorithms

nyalldawson committed:
Move tag handling to QgsProcessingAlgorithm

nyalldawson committed:
Move algorithm icon handling to QgsProcessingAlgorithm

nyalldawson committed:
Move algorithm group handling to QgsProcessingAlgorithm

nyalldawson committed:
Port handling of algorithm names to QgsProcessingAlgorithm

nyalldawson committed:
Move flags (hide from toolbox, etc) to QgsProcessingAlgorithm

nyalldawson committed:
Use formatted name for algorithm names

Avoids conversion of names later and gives clearer separation
between visible names and internal names

3nids committed:
sipify: adds test to prepare-commit and adapt sipdiff

add missing macros to Doxyfile.in

3nids committed:
sipify scripts fixes

* fix prepare-commit.sh when SIP file is not created yet
* raise an error if header doesn't have EXPORT macro for class definition
* sipdiff can generate the SIP file on demand

3nids committed:
fix export macro matching

m-kuhn committed:
More information when [LIB]_EXPORT is missing

m-kuhn committed:
Fix typos

m-kuhn committed:
Autogenerate some sip files

m-kuhn committed:
Improve sip bindings

m-kuhn committed:
mac compatibility fix

m-kuhn committed:
[sipify] Add name of source file to generated headers

m-kuhn committed:
Add whitespace check to prepare-commit.sh

nyalldawson committed:
Fix build when QWebKit not available

nyalldawson committed:
Cleanup QgsRasterDataProvider/QgsRasterLayer progress reporting

Instead of progress changed signals on the provider/layer level,
instead use QgsFeedback arguments for methods which previously
emitted progress changed signals

nyalldawson committed:
Update raster histogram widget to use feedback object

nyalldawson committed:
Handle c++11 '= default' in sipify

nyalldawson committed:
Remove processing sip bindings from sip blacklist

m-kuhn committed:
Fix sipify constructor detection

m-kuhn committed:
Autogenerate more sip files

m-kuhn committed:
Mac compatibility for prepare-commit.sh

m-kuhn committed:
Sipify QgsClipper

m-kuhn committed:
Sipify QgsConditionalStyle

m-kuhn committed:
Fix sipdiff

nyalldawson committed:
Fix(?) build warning

nyalldawson committed:
Remove ingroup and class from doxygen when creating sip docstrings

nyalldawson committed:
Convert doxygen notes and version added to sphinx style in sipify

m-kuhn committed:
Add %Feature support to sipify

nyalldawson committed:
Fix sipify annotations and inline definition combination

nyalldawson committed:
More sipify blacklist removal

nyalldawson committed:
Remove \brief from doxygen in sipify

m-kuhn committed:
Fix sipify override and inline definition combination

m-kuhn committed:
Sipify QgsPropertyTransformer

m-kuhn committed:
Fix build warnings

nyalldawson committed:
Remove SIP_TRANSFERTHIS from .cpp files

m-kuhn committed:
Update doc test for \since

nyalldawson committed:
Replace "python" with "Python" in docs

nyalldawson committed:
QGSI -> QGIS

nyalldawson committed:
Add qgsi->qgis to spell check

m-kuhn committed:
some docstrings in cpp files

nyalldawson committed:
Fix Travis warnings

m-kuhn committed:
Use "\since" instead of "@note added in"

nyalldawson committed:
More new style Qt5 connects

m-kuhn committed:
Add some remaining \since tags

m-kuhn committed:
Update sipify to use the "\since" tag

nyalldawson committed:
Merge pull request #4321 from m-kuhn/sinceNow

Use \since instead of \note added in

nyalldawson committed:
No more old style connects in providers

nyalldawson committed:
Change doxygen @note to \note

NathanW2 committed:
Some more Qt5 signal converts

nyalldawson committed:
"Introduced in..." -> "\since"

nyalldawson committed:
@param -> \param

nyalldawson committed:
@return(s) -> \return(s)

nyalldawson committed:
\return -> \returns (for consistency)

nyalldawson committed:
@see -> \see

nyalldawson committed:
@sa -> \see

nyalldawson committed:
@class -> \class

nyalldawson committed:
@brief -> \brief

nyalldawson committed:
Remove @author credits from doxygen

Because
- not widely used, seems strange that some authors get credit
while others don't
- already included in copyright header

nyalldawson committed:
Remove some unneeded copydocs

Since doxygen already copies docs for overridden methods

nyalldawson committed:
@ingroup -> \ingroup

nyalldawson committed:
@throws -> \throws

nyalldawson committed:
Fix some malformed doxygen

nyalldawson committed:
@(end)link -> \(end)link

nyalldawson committed:
Remove redundant doxygen comments from .cpp files, move to .h
where appropriate

nyalldawson committed:
/*! -> /**

We already have enough doxygen styles to manage!

nyalldawson committed:
Spell check: vecotr -> vector

nyalldawson committed:
Use auto doxygen links where possible instead of explicit \link

Will make things nicer in Python documentation

nyalldawson committed:
Update sip files

nyalldawson committed:
Merge pull request #4308 from nyalldawson/raster_unique

[FEATURE] Allow easy styling of discrete rasters

nyalldawson committed:
Modernise CharacterWidget API, move to sipify

nyalldawson committed:
Strip \a from doxygen comments in sipify

nyalldawson committed:
Update some recently merged classes

nyalldawson committed:
Merge pull request #4323 from nyalldawson/doxygen_standard

Standardise a bunch of doxygen tags, cleanup code

nyalldawson committed:
Handle Q_DECLARE_FLAGS and Q_DECLARE_OPERATORS_FOR_FLAGS in sipify

nyalldawson committed:
Sipify QgsProcessingAlgorithm

nyalldawson committed:
[sipify] Handle enum values using bitshift operators

nyalldawson committed:
Move format handling flags to QgsAlgorithmProvider

nyalldawson committed:
[processing] Move dataobjects.getSupportedOutputVectorLayerExtensions
to QgsVectorFileWriter

nyalldawson committed:
Fix failing test

3nids committed:
[sipify] properly get out of private section

3nids committed:
[sipify] fix override removal

nyalldawson committed:
Move algorithm id (previously commandLineName) to QgsProcessingAlgorithm

nyalldawson committed:
[processing] Fix MessageBarProgress

nyalldawson committed:
Update PreconfiguredAlgorithm for new API

tomkralidis committed:
MetaSearch: set 4326 as default CRS, use WKT to feed polygon geometry

3nids committed:
bump astyle version to 3.0

3nids committed:
[astyle] set C++ version to 11

tomkralidis committed:
Merge pull request #4325 from tomkralidis/metasearch-fixes

MetaSearch: set 4326 as default CRS, use WKT to feed polygon geometry

3nids committed:
[astyle] update licence

tomkralidis committed:
[FEATURE] [needs-docs] MetaSearch: add support for ESRI MapServer/FeatureServer layer integration (fixes #16320)

jef-n committed:
fix api break dox for QgsLayerTreeMapCanvasBridge custom layer order

tomkralidis committed:
Merge pull request #4327 from tomkralidis/issue-16320

[FEATURE] [needs-docs] MetaSearch: add support for ESRI MapServer/FeatureServer layer integration (fixes #16320)

nyalldawson committed:
Update identation

nyalldawson committed:
Complete daily allocation of Qt5 connect transitions

nyalldawson committed:
Move provider algorithm handling to QgsProcessingProvider

nyalldawson committed:
Don't try to deepcopy QgsProcessingAlgorithm class

NathanW2 committed:
Move editor tools into editors folder (#4324)

Move editor tools into editors folder and add test QtCreator template

nyalldawson committed:
Move more algList functionality to QgsProcessingRegistry

nyalldawson committed:
Remove processing algList

All functionality has been moved to QgsProcessingRegistry

nyalldawson committed:
Use display name in algorithm help

nyalldawson committed:
Move tr out of AlgorithmProvider to subclasses

nyalldawson committed:
Fix some breakage

nyalldawson committed:
Remove activation control from AlgorithmProvider

Split between QgsProcessingProvider and subclasses

nyalldawson committed:
Refactor provider actions and remove from AlgorithmProvider

nyalldawson committed:
Also refactor provider context menu actions and remove from AlgorithmProvider

nyalldawson committed:
Remove AlgorithmProvider

Totally ported to QgsProcessingProvider

nyalldawson committed:
Remove processing specific methods for adding/removing providers

QgsProcessingRegistry methods should always be used instead

nyalldawson committed:
Update sip

nyalldawson committed:
Fix processing error when grass7 not available

nyalldawson committed:
Some cleanups

m-kuhn committed:
[sipify] Force newlines after sip commands

m-kuhn committed:
Sipification

m-kuhn committed:
[sipify] do not copy struct docstrings

3nids committed:
fix building without Qt Webkit

also Qt5 connect style

volaya committed:
fixed saga algorithm to adapt it to 2.3 syntax

volaya committed:
[processing] cast setting size value to int

fixes #16347

volaya committed:
[Processing] prevent error when running escapeAndJoin with empty param string

fixes #16200

volaya committed:
Revert "[Processing] prevent error when running escapeAndJoin with empty param string"

This reverts commit c2a566a3a7bfe369832e5361a0c582bfe2ea5a98.

volaya committed:
[Processing] prevent error when running escapeAndJoin with empty param string

fixes #16200

nyalldawson committed:
Merge pull request #4311 from nyalldawson/proc_c4

Simple c++ base class for processing algorithms

nyalldawson committed:
Daily connectification

nyalldawson committed:
[sipify] Replace :: with . during doxygen->docstring conversion

e.g.

You can use QgsXmlUtils::writeVariant to save it to an XML document
->
You can use QgsXmlUtils.writeVariant to load it from an XML document

nyalldawson committed:
Spell check "condtion" -> "condition"

nyalldawson committed:
Allow storing QgsCoordinateReferenceSystem in QVariants

nyalldawson committed:
Port some processing dataobjects routines to c++

nyalldawson committed:
Merge pull request #4331 from nyalldawson/dataobjects

Port some processing dataobjects routines to c++

3nids committed:
[sipify] skip protected members

3nids committed:
[sipify] fix private inheritance with namespace

nyalldawson committed:
Use QgsMapLayerComboBox in processing field calculator

nyalldawson committed:
Port more processing dataobjects methods to c++

Also
- simplify and add tests
- remove large memory leak (persistant store of all non-project layers)
- remove broken support for direct loading postgres/virtual layers
by string (Python version was very broken and would never match
a postgres/virtual layer)

jef-n committed:
creatensis.pl: fix md5sum verification on filenames with a plus

nyalldawson committed:
Merge pull request #4333 from nyalldawson/dataobjects

Port more processing dataobjects methods to c++

gmonfardini committed:
bug 16389 fix error in sld export about stroke color

elpaso committed:
Merge pull request #4334 from gabrimonfa/bug_16389_sld_stroke_color

fixes #16389 stroke color is not correctly exported in SLD

rduivenvoorde committed:
Fix dutch service url

English 'eng' is now a poorly implemented redirect to 'dut', which fails. So fixing this already here while the redirect is fixed at the Kadaster

m-kuhn committed:
[sipify] Add support for abstract classes

m-kuhn committed:
QgsAttributeEditorElement is an abstrat class

alexbruy committed:
add context menu with select/deselect all options

NathanW2 committed:
Add install script for editor templates and code style

NathanW2 committed:
Fix typo in install template script

NathanW2 committed:
Fix unset var in template install script

Pro Tip: Don't code on the train early in the morning

NathanW2 committed:
Fix broken template

Commit:
Fix startup NSBitmapImageRep exceptions on macOS, due to missing files

See https://github.com/qgis/homebrew-qgisdev/issues/27

Exceptions happen when PNGs were changed to SVGs, but the .ui file was
not entirely updated. Then, when PNGs were deleted, a valid, but null,
QIcon gets created for the macOS menu item and the exception is thrown.

Qt 5.8.0 regression.

NathanW2 committed:
Add ignore for astyle for template folder

NathanW2 committed:
Add API area selector in test template

nyalldawson committed:
[FEATURE] Background saving of raster layers

Use the task manager framework to handle saving of raster layers
in the background

nyalldawson committed:
Much faster save as vector layer when using restricted extent

Use a provider filter rect to avoid looping through every single
feature in this case, and also use a prepared extent geometry
to speed up the remaining intersection checks.

borysiasty committed:
Fix Refactor Fields error if layer has no features

borysiasty committed:
Merge branch 'master' of github.com:qgis/QGIS

nirvn committed:
[OGR] fix feature count issue for OSM datasets (fixes #16402) (#4322)

timlinux committed:
Added more error handling for cases when processing help causes an error

timlinux committed:
Changed Tile Server (XYZ) to XYZ Tile Server to reduce visual clutter

nyalldawson committed:
Merge pull request #4335 from nyalldawson/save_vector_with_extent

Much faster save as vector layer when using restricted extent

nyalldawson committed:
Merge pull request #4336 from nyalldawson/background_raster

[FEATURE] Background saving of raster layers

nyalldawson committed:
Merge pull request #4337 from timlinux/browser

Changed Tile Server (XYZ) to XYZ Tiles to reduce visual clutter

Gustry committed:
disable export button if we try to export a raster layer (#4319)

nyalldawson committed:
Update api break docs, sipify QgsRasterFileWriter

nyalldawson committed:
[sipify] Replace nullptr with None in docstrings

Since nullptr has no meaning to Python devs

nyalldawson committed:
Whitespace

nyalldawson committed:
Fix error when trying to run processing algs

Caused by trying to deepcopy a c++ object. This is a temporary
fix - when the next part of the processing/c++ work lands algorithm
copies will no longer be required

nyalldawson committed:
[sipify] Add return types for functions in docstrings

and change \return -> :return: in docystrings

nyalldawson committed:
Fix bad string encoding on windows

nyalldawson committed:
Always return const algorithms, since algorithms should not be modified

nyalldawson committed:
Attempt to fix failing GDAL test

nyalldawson committed:
Indentation

nyalldawson committed:
Merge pull request #4328 from nyalldawson/proc_c7

Porting processing to c++, part 2 (followup to #4311)

alexbruy committed:
fix indentation

alexbruy committed:
Merge pull request #4338 from timlinux/processing-fixes

[processing] Added more error handling for cases when processing help causes an error

alexbruy committed:
remove unused label from about dialog

nyalldawson committed:
Avoid showing "0.000 deg" in measure dialog for small degree measurements

Instead auto add decimal places so that a usable value is shown
when the unit is degrees and the setting for decimal places isn't
sufficient to show a value.

m-kuhn committed:
sipify_all

m-kuhn committed:
Sipify

alexbruy committed:
don't close popup after first click, so user can select all necessary
options at once

alexbruy committed:
sipify

alexbruy committed:
homogenize doxygen comments

m-kuhn committed:
[sipify] Fix exposing members and default struct accessors

alexbruy committed:
replace "added in QGIS 3.0" with "since QGIS 3.0"

alexbruy committed:
fix warnings`

m-kuhn committed:
[postgis] Native boolean support

m-kuhn committed:
CheckBox widget: support native Bool fields

m-kuhn committed:
Add test for postgres boolean

NathanW2 committed:
Add header to Qt templates

nyalldawson committed:
Merge pull request #4339 from nyalldawson/sipify

Sipify

alexbruy committed:
address Nyall's comments:
 - add TransferThis to parent argument
 - QSignalSpy always available with Qt5

m-kuhn committed:
Fix sipify test header file

alexbruy committed:
update SIP file

m-kuhn committed:
[sipify] skip non-public template based member declarations

m-kuhn committed:
Some docstring improvements

alexbruy committed:
remove docs for parent parameter

NathanW2 committed:
sip skip on ColumnConfig == for now

Fails on Windows build for some reason

NathanW2 committed:
Test cmake (#4343)

* Streamline cmake test file adding

NathanW2 committed:
Remove old comment line

[ci skip]

nyalldawson committed:
Fix missing sip cast for QgsHillshadeRenderer

nyalldawson committed:
Add missing currentThemeChanged signal to interface

nyalldawson committed:
Daily connectification

nyalldawson committed:
Rename QgsMapLayer::metadata -> htmlMetadata()

Frees up metadata() for use as a structured metadata getter

alexbruy committed:
Merge pull request #4299 from alexbruy/checkable-combobox

[FEATURE] New custom widget checkable combobox

nyalldawson committed:
[sipify] Also skip mutable protected members

nyalldawson committed:
Sipify QgsMapLayer

nyalldawson committed:
Remove QgsMapLayer::invalidTransformInput slot

Calling slot does nothing and is not used anywhere

nyalldawson committed:
Fix some QgsMapLayer getters which are incorrectly marked as slots

nyalldawson committed:
Avoid some "" literals and QString::nulls in layer API

Use QString() instead

nyalldawson committed:
Merge pull request #4348 from nyalldawson/sipify_maplayer

Sipify QgsMapLayer and cleanup API

mj10777 committed:
UTF8 support for QgsRasterDataProvider::cStringList2Q_ (#4345)

UTF8 support for QgsRasterDataProvider::cStringList2Q_.

Fix #15978

nyalldawson committed:
Merge pull request #4346 from nyalldawson/metadata_api

Rename QgsMapLayer::metadata() -> htmlMetadata()

nyalldawson committed:
Use weak pointers for registered QgsOptionsWidgetFactorys

Avoids QGIS crashing if a plugin crashes without deregistering
its QgsOptionsWidgetFactory.

nyalldawson committed:
Make QgsRectangle protected members private

This class is not designed to be subclassed

nyalldawson committed:
[sipify] Fix stripping function body with virtual const reference returns

ghtmtt commented:
@m-kuhn what do you think about this PR? even without tests, could we merge it (after having resolved the conflicts)?

m-kuhn commented on python/plugins/processing/algs/qgis/BoxPlot.py:
```diff
@@ -0,0 +1,91 @@
+                                              self.INPUT,
+                                              ParameterTableField.DATA_TYPE_NUMBER))
+        self.addParameter(ParameterBoolean(self.MSD,
+                                           self.tr('Show also standard deviation'), False))
```
Should that be a `ParameterSelection` with options `True`, `False`, `'sd'`?

ghtmtt replied:
yes I should be.. I tried to follow some *templates* but it turn out that when looping on the `ParameterSelection` within the script, then I get the indexes of the parameters and not the parameters itself (needed by `plotly`). So in the meanwhile to look for something smarter I was thinking to let the user choose between `yes` or `no`

m-kuhn replied:
Just do

```
msdIndex = self.getParameterValue(self.MSD)
msd = True
if msdIndex == 1:
    msd = 'sd'
elif msdIndex == 2:
   msd = False
```

Or if you want to do it clean and fancy, define an enum somewhere on top of your file

```
MSD_SD, MSD_ON, MSD_OFF = range(3)
```

and then use

```
if msdIndex == MSD_ON:
```

etc.

------------

Sidenote, I would skip the word `also` - and simply say "Show standard deviation"

m-kuhn commented on python/plugins/processing/algs/qgis/BoxPlot.py:
```diff
@@ -0,0 +1,91 @@
+                    y=values[valuefieldname],
+                    boxmean='sd')]
+
+        plt.offline.plot(data, filename=output, auto_open=False)
```
Would it be interesting to add `auto_open` as a parameter?

ghtmtt replied:
that is also the default parameter.. but I saw that the first plots added bu @alexbruy have the `auto_open` set to False. If also Alex agrees, I can add a checkbox (maybe default checked?!) with the `auto_open` option

m-kuhn commented:
Sure, this is mergable without tests.
Just ping me when you have it rebased (and handled the comments if you think they make sense).

ghtmtt committed:
resolve conflits for new plots

ghtmtt committed:
combobox for Boxplot statistics

ghtmtt commented:
ok, done ;) thanks for the hints of the `combobox` in Processing.. 
I rebased and fixed the damns conflicts... hopefully travis won't complain

Thanks again @m-kuhn

m-kuhn commented:
Whoops, look like something has gone a bit weird here...

ghtmtt commented:
@m-kuhn as always :( ... could you please trash this PR? I made a new one from scratch #4359 cleaner and simpler..

m-kuhn commented:
If you paste here what you tried, I can try to tell you where things went wrong...

ghtmtt commented:
I thought it could be simple...
* git checkout master
* git pull upstream master
* git checkout proc_plotly
* git rebase master

fixing some conflicts and then just as always (git add, git commit.....).

But don't waste your time...

m-kuhn commented:
I guess your local master is somehow not in sync with the upstream master.

To bypass the problem (next time) use this instead:

```
git checkout proc_plotly
git fetch upstream
git rebase upstream/master
```

Then repeat

```
git mergetool
git rebase --continue
# Or sometimes ...
git rebase --skip
```

ghtmtt commented:
thanks! Next time I'll try this ;)
